### ISSUE TITLE
useAsyncDispatch returns a new reference every time it's called.

### ISSUE THREAD

**Issue by harm-meijer** [commercetools/merchant-center-application-kit#2818]
Consider the following line:

`const dispatch = useAsyncDispatch();`

When using dispatch in an effect it will add `dispatch` to the dependencies:

```
  useEffect(() => {
    dispatch(...)
  }, []);//dispatch is a dependency
```

Because useAsyncDispatch returns a new reference every time it's called you either cannot add it to the dependency list or will cause an infinite loop:

```
import { useEffect, useState } from 'react';
import { useAsyncDispatch, actions } from '@commercetools-frontend/sdk';

let lastValue = undefined;
const Component = () => {
  const [data, setData] = useState();
  const dispatch = useAsyncDispatch();
  if (!lastValue) {
    console.log('...setting last value');
    lastValue = dispatch;
  }
  useEffect(() => {
    async function execute() {
      try {
        const result = await dispatch(
          //docs: https://docs.commercetools.com/custom-applications/development/data-fetching#rest-http-client
          actions.forwardTo.get({
            uri: 'uri',
          })
        );
        setData(result);
        // Update state with `result`
      } catch (error) {
        // Update state with `error`
      }
    }
    execute();
  }, []); //cannot add dispatch (see log below)
  if (!(lastValue === dispatch)) {
    console.log('...setting dependency to dispatch will cause infinite loop');
  }
  return <pre>{JSON.stringify(data, undefined, 2)}</pre>;
};
```

I am unable to find the code on github (npm points to this repo but useAsyncDispatch is not defined in this repo). In node_modules/@commercetools-frontend/sdk/dist/commercetools-frontend-sdk.esm.js I can find the following:

```
function useAsyncDispatch() {
  var dispatch = useDispatch();
  return function (action) {
    return dispatch(action);
  };
}
```

This should be:

```
function useAsyncDispatch() {
  var dispatch = useDispatch();
  return useCallback(
    function (action) {
      return dispatch(action);
    },
    [dispatch]
  );
}
```

**Comment by emmenko** [comment]
You're right, good catch!! Thanks for letting us know.

I'll open a PR with a fix.

**Comment by emmenko** [comment]
@harm-meijer The fix will be available in the next release `v21.15.0`. Thanks again for looking into that and apologies for the trouble.
